testing-tools/testing-tools-macros/src/randomizer.rs

```diff
@@ -0,0 +1,171 @@
+    }
+}
+
+fn derive_enum(item: DataEnum, name: Ident) -> proc_macro2::TokenStream {
+    let varient_matches = item
+        .variants
+        .iter()
+        .map(|variant| enum_variant_match(&name, &variant.ident, &variant.fields))
+        .collect::<Vec<_>>();
+
+    let varient_names = item
+        .variants
+        .into_iter()
+        .map(|variant| variant.ident)
+        .collect::<Vec<_>>();
+
+    let any_variant = any_enum_variant(&name, varient_names);
+
+    let matches = quote::quote! {
+        #(#varient_matches)*
+        #any_variant
+    };
+
+    let macro_name = Ident::new(&format!("random_{}", name.to_string()), Span::call_site());
+    let mod_name =
+        Ident::new(&format!("mod_{}", name.to_string().to_lowercase()), Span::call_site());
+    quote::quote! {
+        pub use #mod_name::#macro_name;
+        #[macro_use]
+        mod #mod_name {
+            #[macro_export]
+            macro_rules! #macro_name {
+                #matches
+            }
+            pub use #macro_name;
+        }
+    }
+}
+
+fn enum_variant_match(
+    enum_name: &Ident,
+    variant_name: &Ident,
+    variant_fields: &Fields
+) -> proc_macro2::TokenStream {
+    if variant_fields.len() == 0 {
+        quote::quote! {
+            (#variant_name) => {
+                {
+                    #enum_name::#variant_name
+                }
+            };
+        }
+    } else if variant_fields.len() == 1 {
+        let ty = variant_fields.into_iter().next().unwrap().ty.clone();
+        let field = match ty {
+            // Type::Paren(type_paren) => type_paren,
+            // Type::Array(type_array) => todo!(),
+            // Type::BareFn(type_bare_fn) => todo!(),
+            // Type::Group(type_group) => todo!(),
+            // Type::ImplTrait(type_impl_trait) => todo!(),
+            // Type::Infer(type_infer) => todo!(),
+            // Type::Macro(type_macro) => todo!(),
+            // Type::Never(type_never) => todo!(),
+            Type::Path(type_path) => type_path.path.get_ident().cloned().unwrap(),
+            // Type::Ptr(type_ptr) => todo!(),
+            // Type::Reference(type_reference) => todo!(),
+            // Type::Slice(type_slice) => todo!(),
+            // Type::TraitObject(type_trait_object) => todo!(),
+            // Type::Tuple(type_tuple) => todo!(),
+            // Type::Verbatim(token_stream) => todo!(),
+            _ => unreachable!("type is {:?}", ty)
+        };
+
+        quote::quote! {
+            (#variant_name | $(($field_name:ident, $value:expr)),*) => {
+                {
+                    //#random_field!($($field_name:ident, $value:expr),*)
+                    let mut rng = rand::thread_rng();
+                    let mut this: #field = rand::Rng::gen(&mut rng);
+                    $(
+                        this.$field_name = $value;
+                    )*
+                    this
+                }
+            };
```

Comment:
just have it generate a builder for this. 100 times more readable with a clean interface rather than a macro thats generated by a macro. if someone tried to look at this to see how they would use this macro. They would have find this exact file. this is extremely degenerate.